mapbox_gl_platform_interface/lib/src/mapbox_gl_platform_interface.dart

```diff
@@ -51,269 +45,77 @@ abstract class MapboxGlPlatform {
 
   final onUserLocationUpdatedPlatform = ArgumentCallbacks<UserLocation>();
 
-  Future<void> initPlatform(int id) async {
-    throw UnimplementedError('initPlatform() has not been implemented.');
-  }
-
+  Future<void> initPlatform(int id);
   Widget buildView(
       Map<String, dynamic> creationParams,
       OnPlatformViewCreatedCallback onPlatformViewCreated,
-      Set<Factory<OneSequenceGestureRecognizer>>? gestureRecognizers) {
-    throw UnimplementedError('buildView() has not been implemented.');
-  }
-
-  Future<CameraPosition?> updateMapOptions(
-      Map<String, dynamic> optionsUpdate) async {
-    throw UnimplementedError('updateMapOptions() has not been implemented.');
-  }
-
-  Future<bool?> animateCamera(CameraUpdate cameraUpdate) async {
-    throw UnimplementedError('animateCamera() has not been implemented.');
-  }
-
-  Future<bool?> moveCamera(CameraUpdate cameraUpdate) async {
-    throw UnimplementedError('moveCamera() has not been implemented.');
-  }
-
+      Set<Factory<OneSequenceGestureRecognizer>>? gestureRecognizers);
+  Future<CameraPosition?> updateMapOptions(Map<String, dynamic> optionsUpdate);
+  Future<bool?> animateCamera(CameraUpdate cameraUpdate);
```

Comment:
This change means that if we ever add support for a new platform (e.g. desktop) that new platform must implement animateCamera, correct?

Reply:
Yeah but the implementation can throw in the same way as here. 
The reason for this change is that with the old style it was super easy to miss one method on one platform. Much better to just let the compiler tell you what you still have to implement.

Reply:
Good point, and since new platforms would probably support at least most of the methods it shouldn't be too much work to manually throw for the missing ones.